### ISSUE TITLE
run_once[task] incorrectly fails

### ISSUE THREAD

**Issue by jkossis** [ansible/ansible-lint#2815]
##### Summary

The run_once linting rule incorrectly fails when using run_once within a task, along with the default playbook strategy (linear).

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```
ansible-lint 6.9.1 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

The subsequent code block yields this ansible-lint error: 

`Using run_once may behave differently if strategy is set to free.ansible-lint[run_once[task]](https://ansible-lint.readthedocs.io/rules/run-once/)`

```
- name: This is a test playbook
  hosts: all
  strategy: linear
  tasks:
    - name: This is a test task
      ansible.builtin.debug:
        msg: "Hello World!"
      run_once: true
```

##### Desired Behavior

That code block above should be valid, as the strategy is set to `linear` (and not `free`). 

##### Actual Behavior

It seems pretty straightforward as to why this is occurring. When taking a look at the `matchtask` function added [here](https://github.com/ansible/ansible-lint/pull/2626/files#diff-3177d7c4deb1f71b95aafee3cde94e887987007d8862080f1b55cb85d9cf5f34), you can see that it only checks for `run_once`, and doesn't take into account the strategy (like its sibling `matchplay` function does). Additionally, the examples/tests included in that PR don't cover this case. They only cover not using `run_once` at all, or using it with the `free` strategy.

**Comment by ssbarnea** [comment]
The linter will never be able to identify how the user will use this task in production, mainly because the strategy can be altered at runtime. The only way to silence this is to make use of `noqa` to silence it. That is perfectly normal and acts as "I know the risks, but I decide to ignore them.".

**Comment by jkossis** [comment]
Taking what you mentioned for granted, this rule is likely misnamed/wrongly described then.

It essentially means you cannot use `run_once` at the task level, regardless of the play's strategy (`free` or otherwise).

**Comment by ssbarnea** [comment]
You can, see https://ansible-lint.readthedocs.io/usage/#muting-warnings-to-avoid-false-positives -- it is an expected false-positive, and you just need to add a comment to silence it.

**Comment by jkossis** [comment]
I do understand how to make use of `noqa` for silencing purposes. That said, I still contest that this isn't really a false positive. The description of the rule (don't use `run_once` with strategy `free`) doesn't match the implementation of the rule (don't use `run_once` with any strategy).

An example of a false positive, IMO, is if a line were 200 characters long, but I really wanted it to be that way. The rule dictating that (lines should be less than 160) is still accurately described.

**Comment by ssbarnea** [comment]
@shatakshiiii If you can please update the md file and include these:
- warning block via ```{warning}...``` mentioning that rule will always trigger regardless which values they configure inside `strategy` field. That is be because the effective value used at runtime can be different than the value inside the file (ansible command line arguments)
- Say that is perfectly ok to add `# noqa: run_once[task]` to mark the warning as acknowledged and ignored
- Include an example of noqa comment use as alternative solution

Basically this rule exists to remind user that run_once is more of try_to_run_once, and that there is no guarantee that it will not run more than once.

Some people are not aware of this and add run_tasks to things would would produce problems when ran multiple times.
